sines.lua

```diff
@@ -116,6 +146,69 @@ function cleanup()
   clock.cancel(redraw_clock)
 end
 
+--scala stuff
+
+-- set the root frequency, without changing root note
+-- this effects a transposition
+function set_root_frequency(freq)
+  tuning_state.root_freq = freq
+  params:set('root_freq', freq, true)
+end
+
+-- set the current tuning, by ID string
+function set_tuning(key)
+  print('key iiiiiiiiiiiiiiii' .. key)
+  tuning_name = tuning_keys[key]
+  tuning = tuning_scala.load_file(_path.dust .. 'code/sines/lib/tunings/'.. tuning_key .. '.scl')
+  print('urrrgh: ')
+  tab.print(tuning)
+  tuning_state.selected_tuning = key
+  params:set('tun', key, true)
+end
+
+function set_tuning_proper(d)
+  local sel = tuning_state.selected_tuning
+  local i = tuning_keys_rev[sel]
+  i = util.clamp(i + d, 1, num_tunings)
+  set_tuning_index(i)
+end
+
+function build_tuning_keys_reversed()
+   table.sort(tuning_keys)
+   tuning_keys_rev = {}
+   for i, v in ipairs(tuning_keys) do
+      tuning_keys_rev[v] = i
+   end
+end
+
+-- set the current tuning, by numerical index
+function set_tuning_index(idx)
+  print(tuning_keys[idx])
+  --WHY DOESN'T THIS WORK AARGH
```

Comment:
Something silly in lua about concatenating strings and vars in parens? urgh

Reply:
https://stackoverflow.com/a/21792185/7010427

Reply:
Something like:

local t = {}
for i = 1, 1000 do
    table.insert(t, tostring(i))
end
local a = table.concat(t) -- "1234...9991000"